lib/js_of_ocaml/performanceObserver.mli

```diff
@@ -0,0 +1,59 @@
+(** PerformanceObserver API
+
+    A code example:
+    {[
+      if (PerformanceObserver.is_supported()) then
+        let entry_types = [ "measure" ] in
+        let f entries observer =
+          let entries = entries##getEntries in
+          Firebug.console##debug entries ;
+          Firebug.console##debug observer
+        in
+        PerformanceObserver.observe ~entry_types ~f
+        ()
+   ]}
+
+   @see <https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver> for API documentation.
+*)
+
+class type performanceObserverInit =
+  object
+    method entryTypes : Js.js_string Js.t Js.js_array Js.t Js.writeonly_prop
+  end
+
+class type performanceEntry =
+  object
+    method name : Js.js_string Js.t Js.readonly_prop
+
+    method entryType : Js.js_string Js.t Js.readonly_prop
+
+    method startTime : float Js.readonly_prop
+
+    method duration : float Js.readonly_prop
+  end
+
+class type performanceObserverEntryList =
+  object
+    method getEntries : performanceEntry Js.t Js.js_array Js.t Js.meth
+  end
+
+class type performanceObserver =
+  object
+    method observe : performanceObserverInit Js.t -> unit Js.meth
+
+    method disconnect : unit Js.meth
+
+    method takeRecords : performanceEntry Js.t Js.js_array Js.t Js.meth
+  end
+
+val performanceObserver :
+  (   (performanceObserverEntryList Js.t -> performanceObserver Js.t -> unit) Js.callback
+   -> performanceObserver Js.t)
+  Js.constr
+
+val is_supported : unit -> bool
```

Comment:
i wonder if `is_supported` should be baked in, or left to user space? genuine ask, not a passive challenge :)

Reply:
What do you mean by backed it ?

Reply:
sorry, it's a quirky phrase :)

> baked in: incorporate something as an integral part of a product, service, or system.

one may observe that https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/isSupported doesn't exist. thus, i am asking "should jsoo decorate web apis with extra functions, methods, or properties?"

Reply:
I've included `is_supported` for consistency with the existing `intersectionObserver` and `resizeObserver` APIs. One could quite reasonably argue that jsoo should just expose web APIs and nothing more, and I might even agree with it. However, I think this PR does the correct thing by following the existing convention.